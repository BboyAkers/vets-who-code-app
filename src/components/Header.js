import React, { useRef, Fragment } from 'react'
import Slider from 'react-slick'
import { Link } from 'gatsby'
import BackgroundSection from '../components/BackgroundSection'

const settings = {
  infinite: true,
  speed: 500,
  fade: true,
  lazyLoad: true,
  slidesToShow: 1,
  slidesToScroll: 1,
  autoplay: true,
  autoplaySpeed: 3000,
  nextArrow: <Fragment />, // use Fragment to disable react-slick bd
  prevArrow: <Fragment />,
}

function Header() {
  const sliderRef = useRef()

  return (
    <header id="site-header" className="site-header flexslider classic">
      <div style={{ backgroundColor: 'rgb(9, 31, 64)' }}>
        <Slider {...settings} ref={sliderRef}>
          <BackgroundSection fileName="code.jpg" critical fadeIn>
            <div className="header-classic wrapper-table overlay-01">
              <div className="valign-center">
                <div className="container">
                  <div className="col-md-10 col-md-offset-1">
                    <div className="intro text-left">
                      <h1>Learn</h1>
                      <p className="subtitle">How To Code With Other Veterans.</p>
                      <div className="btn-cal-group">
                        {' '}
                        <Link to="/apply" className="btn btn-charity-default">
                          Apply
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </BackgroundSection>
          <BackgroundSection fileName="speaking.jpg">
            <div className="header-classic  wrapper-table overlay-01">
              <div className="valign-center">
                <div className="container">
                  <div className="col-md-10 col-md-offset-1">
                    <div className="intro text-left">
                      <h1>Sign Up</h1>
                      <p className="subtitle">As A Mentor.</p>
                      <div className="btn-cal-group">
                        {' '}
                        <Link to="/mentor" className="btn btn-charity-default">
                          {' '}
                          Sign Up
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </BackgroundSection>
          <BackgroundSection fileName="this_is_us.jpg">
            <div className="header-classic wrapper-table overlay-01">
              <div className="valign-center">
                <div className="container">
                  <div className="col-md-10 col-md-offset-1">
                    <div className="intro text-right">
                      <h1 className="red">
                        <span className="red">Help Us</span>
                      </h1>
                      <p className="subtitle red">Teach More Veterans How To Code.</p>
                      <div className="btn-cal-group">
                        {' '}
                        <Link to="/donate" className="btn btn-charity-default">
                          Please Donate
                        </Link>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </BackgroundSection>
        </Slider>
        <ol className="flex-control-nav flex-control-paging">
          <li>
            <a className="">1</a>
          </li>
          <li>
            <a className="flex-active">2</a>
          </li>
          <li>
            <a className="">3</a>
          </li>
        </ol>
        <ul className="flex-direction-nav">
          <li className="flex-nav-prev" onClick={() => sliderRef.current.slickNext()}>
            <a className="flex-prev" href="#">
              Previous
            </a>
          </li>
          <li className="flex-nav-next" onClick={() => sliderRef.current.slickPrev()}>
            <a className="flex-next" href="#">
              Next
            </a>
          </li>
        </ul>
      </div>
    </header>
  )
}
export default Header
