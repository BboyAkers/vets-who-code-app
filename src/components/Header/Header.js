import React from 'react'
import { graphql, useStaticQuery, Link } from 'gatsby'
import BackgroundImage from 'gatsby-background-image'
import Typed from 'react-typed'
import Toggle from '../Toggle'
import '../../assets/css/custom.css'
function Header() {
  const data = useStaticQuery(graphql`
    query {
      codeImage: file(relativePath: { eq: "code.jpg" }) {
        childImageSharp {
          fluid {
            ...GatsbyImageSharpFluid_withWebp_noBase64
          }
        }
      }
    }
  `)

  return (
    <BackgroundImage
      Tag="section"
      className="site-header flexslider classic overlay main-overlay grey"
      backgroundColor="#091f40"
      fluid={data.codeImage.childImageSharp.fluid}
      style={{ height: '80vh' }}
      loading="eager"
    >
      <div className="header-classic wrapper-table">
        <div className="valign-center">
          <div className="container">
            <div className="col-md-10 col-md-offset-1">
              <div className="intro text-left" style={{ color: '#fff' }}>
                <h1>
                  <Typed
                    className="typedString"
                    strings={['Learn Javascript', 'Network', 'Get A Job']}
                    typeSpeed={70}
                    backSpeed={80}
                    smartBackspace
                    loop
                  />
                </h1>
                <p className="subtitle">With Vets Who Code.</p>
                <div className="btn-cal-group">
                  <Link to="/apply" className="btn btn-charity-default">
                    Apply
                  </Link>
                  &nbsp;
                  <Link to="/donate" className="btn btn-charity-default">
                    Donate
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </BackgroundImage>
  )
}

export default Header
