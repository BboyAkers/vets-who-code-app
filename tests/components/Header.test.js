import React from 'react'
import Header from '../../src/components/Header'
import { render } from 'react-testing-library'

describe('<Header />', () => {
  test('should render correctly', () => {
    const { container } = render(<Header />);
    expect(container.firstChild).toMatchSnapshot()
  })
})
