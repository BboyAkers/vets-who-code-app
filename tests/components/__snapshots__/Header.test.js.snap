// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<header
  class="site-header flexslider classic"
  id="site-header"
>
  <div>
    <div />
    <ol
      class="flex-control-nav flex-control-paging"
    >
      <li>
        <a
          class=""
        >
          1
        </a>
      </li>
      <li>
        <a
          class="flex-active"
        >
          2
        </a>
      </li>
      <li>
        <a
          class=""
        >
          3
        </a>
      </li>
    </ol>
    <ul
      class="flex-direction-nav"
    >
      <li
        class="flex-nav-prev"
      >
        <a
          class="flex-prev"
          href="#"
        >
          Previous
        </a>
      </li>
      <li
        class="flex-nav-next"
      >
        <a
          class="flex-next"
          href="#"
        >
          Next
        </a>
      </li>
    </ul>
  </div>
</header>
`;
